File: backend/src/routes/modalidades.js
const router = require('express').Router();
const auth = require('../middleware/auth');
const supabase = require('../db');

// GET /modalidades
router.get('/', auth, async (req, res) => {
  try {
    const { ativo } = req.query;
    let q = supabase.from('modalidades').select('*').order('nome');
    if (ativo !== undefined) q = q.eq('ativo', ativo !== 'false');
    const { data, error } = await q;
    if (error) return res.status(400).json({ error: error.message });
    res.json(data);
  } catch (e) { res.status(500).json({ error: e.message }); }
});

// GET /modalidades/:codigo
router.get('/:codigo', auth, async (req, res) => {
  try {
    const { data, error } = await supabase.from('modalidades').select('*').eq('codigo', req.params.codigo).single();
    if (error) return res.status(404).json({ error: 'Modalidade não encontrada' });
    res.json(data);
  } catch (e) { res.status(500).json({ error: e.message }); }
});

// PATCH /modalidades/:id — atualizar cotação/limite
router.patch('/:id', auth, async (req, res) => {
  try {
    const { data, error } = await supabase.from('modalidades').update(req.body).eq('id', req.params.id).select().single();
    if (error) return res.status(400).json({ error: error.message });
    res.json(data);
  } catch (e) { res.status(500).json({ error: e.message }); }
});

module.exports = router;

File: backend/app/routes/modalidades.py
from fastapi import APIRouter, Depends, HTTPException
from typing import Optional
from app.db import get_supabase
from app.auth import get_current_user

router = APIRouter()

@router.get("/")
async def list_modalidades(ativo: Optional[str] = None, user=Depends(get_current_user)):
    q = get_supabase().table("modalidades").select("*").order("nome")
    if ativo is not None:
        q = q.eq("ativo", ativo != "false")
    return q.execute().data

@router.get("/{codigo}")
async def get_modalidade(codigo: str, user=Depends(get_current_user)):
    res = get_supabase().table("modalidades").select("*").eq("codigo", codigo).single().execute()
    if not res.data:
        raise HTTPException(status_code=404, detail="Modalidade não encontrada")
    return res.data

@router.patch("/{modalidade_id}")
async def update_modalidade(modalidade_id: int, body: dict, user=Depends(get_current_user)):
    res = get_supabase().table("modalidades").update(body).eq("id", modalidade_id).execute()
    return res.data[0] if res.data else {}
